python/dgl/transform.py

```diff
@@ -180,9 +180,12 @@ def knn_graph(x, k, algorithm='bruteforce-blas', dist='euclidean'):
             x_seg = x_size[0] * [x_size[1]]
         else:
             x_seg = [F.shape(x)[0]]
-        out = knn(x, x_seg, x, x_seg, k, algorithm=algorithm, dist=dist)
-        row, col = out[1], out[0]
-        return convert.graph((row, col))
+        if algorithm == 'nn-descent':
+            return nndescent_knn_graph(x, k, x_seg, dist=dist)
```

Comment:
Why not wrap in `knn` as well?

Reply:
Because the interface of nn-descent is different. Other KNN algorithms can accept a different point set as the query set. Almost all KNN algorithms need to build an index before query. For example, the KD tree needs to build a tree before querying k-nearest neighbors, and the index of brute-force implementations is just the point set itself. For nn-descent, the index is just the KNN-graph of the input point set. So we only need the index-building process for nn-descent. If we also want the query-different-point-set feature for nn-descent, we can warp nn-descent in `knn` as well.

Reply:
For `knn`, can we make the `y_seg` as `None` by default?

Reply:
Yes, we can. And if we want to warp nn-descent as well, we also need the query-different-point-set implementation for nn-descent here.

Reply:
We can leave it for future work at this point. For DGL use cases, most KNN are self-queried.

Reply:
I just realized my above comment is ambiguous. We can leave 'query-different-point-set implementation for nn-descent' as future work. For now, maybe a unified `knn` interface is still preferred, i.e. wrap `nndescent_knn_graph` into `knn`.

Reply:
Done. Now `y` and `y_segs` can be None if we want a self-query over `x`.